### DISCUSSION TITLE
Bun v1.0

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#4580] [Announcements]
Today is an exciting day.

To install Bun v1.0:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0:
```bash
bun upgrade
```

**[Read Bun v1.0.0's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0)**

## What's Changed
* Fix assertion failure with sourcemaps in multi-threaded transpiler by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4321
* Automatically hot reload Bun.serve() by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4344
* Fix the crash when importing a module that does not exist. by @Hanaasagi in https://github.com/oven-sh/bun/pull/4348
* More improvements to debugger support by @Electroid in https://github.com/oven-sh/bun/pull/4345
* Set the timezone in spawn in tests by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4337
* @electroid's dap changes by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4367
* Fix some edge cases in the `env` param of `spawn`. by @Hanaasagi in https://github.com/oven-sh/bun/pull/4364
* just kernel32 things by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4354
* make pending_response and metdata life cycle more clear and make fetch more reliable by @cirospaciari in https://github.com/oven-sh/bun/pull/4331
* Bring uSockets & uWebSockets forks into Bun's repository by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4372
* remove uws from .gitmodules by @prabhatexit0 in https://github.com/oven-sh/bun/pull/4374
* use `options.fd` if provided for `fs.Read/WriteStream` by @dylan-conway in https://github.com/oven-sh/bun/pull/4378
* feat(node:dns): implement `dns.reverse`. by @Hanaasagi in https://github.com/oven-sh/bun/pull/4332
* fix #4356 by @dylan-conway in https://github.com/oven-sh/bun/pull/4386
* Fix close callback and open event for fs streams by @dylan-conway in https://github.com/oven-sh/bun/pull/4384
* add unix socket support for `node:http.server` by @Hanaasagi in https://github.com/oven-sh/bun/pull/4390
* fix(node-fetch): use stream.Readable instead of web streams by @paperdave in https://github.com/oven-sh/bun/pull/4394
* Fix assertion failure in spawn-related tests by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4400
* More support for DAP by @Electroid in https://github.com/oven-sh/bun/pull/4380
* reset signal handlers in Bun.spawn  by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4405
* fix(http/https) disable decompress on http/https client by @cirospaciari in https://github.com/oven-sh/bun/pull/4399
* minor rebuild diffs by @alexlamsl in https://github.com/oven-sh/bun/pull/4416
* exit handlers for resetting tty by @dylan-conway in https://github.com/oven-sh/bun/pull/4419
* [install] fix stale root life-cycle script in lockfile by @alexlamsl in https://github.com/oven-sh/bun/pull/4411
* add life cycle scripts to lockfile metahash by @dylan-conway in https://github.com/oven-sh/bun/pull/4420
* feat(node:fs): add `cp`/`cpSync`/`promises.cp` + async `copyFile` by @paperdave in https://github.com/oven-sh/bun/pull/4340
* Fix(semver): fix matching with pre-release version by @Hanaasagi in https://github.com/oven-sh/bun/pull/4412
* get name if not provided in `FormData.append` by @dylan-conway in https://github.com/oven-sh/bun/pull/4434
* `bun install` correctly join dependency URLs by @Parzival-3141 in https://github.com/oven-sh/bun/pull/4421
* only set initial debugger breakpoint once by @dylan-conway in https://github.com/oven-sh/bun/pull/4441
* fix(runtime): `fs.cp` edge cases by @paperdave in https://github.com/oven-sh/bun/pull/4439
* fix(runtime): fix dns_resolver crash by @paperdave in https://github.com/oven-sh/bun/pull/4435
* fix `Bun.serve` with tls and `Bun.file` by @dylan-conway in https://github.com/oven-sh/bun/pull/4450
* keep `export * as` by @dylan-conway in https://github.com/oven-sh/bun/pull/4451
* Fix typo in docs/api/tcp by @JorgeJimenez15 in https://github.com/oven-sh/bun/pull/4445
* docs: fix http simple example log statement by @karlbohlmark in https://github.com/oven-sh/bun/pull/4320
* fix(runtime): Fix `fs.stat().mtime` on MacOS by @paperdave in https://github.com/oven-sh/bun/pull/4440
* fix(syscall): fix handling syscall errno by @Hanaasagi in https://github.com/oven-sh/bun/pull/4461
* fix(HTMLRewriter) buffer response before transform by @cirospaciari in https://github.com/oven-sh/bun/pull/4418
* docs typo: `loaders` -> `loader` by @Hanaasagi in https://github.com/oven-sh/bun/pull/4476
* fix zlib deflate on fetch by @cirospaciari in https://github.com/oven-sh/bun/pull/4483
* remove `chmod` usage with `--bun` flag by @dylan-conway in https://github.com/oven-sh/bun/pull/4490
* fix dup syscall on Windows by @CGQAQ in https://github.com/oven-sh/bun/pull/4496
* minor rebuild diffs by @alexlamsl in https://github.com/oven-sh/bun/pull/4486
* fix ipv6 localhost fetch by @dylan-conway in https://github.com/oven-sh/bun/pull/4498
* fix checkout/build failure due to `src/deps/uws` by @alexlamsl in https://github.com/oven-sh/bun/pull/4505
* fix(install): ensure all lockfile structs do not have undefined padding by @paperdave in https://github.com/oven-sh/bun/pull/4401
* fix(fetch) fix proxy tunneling by @cirospaciari in https://github.com/oven-sh/bun/pull/4510
* fix(fetch) always use readable stream if it is available by @cirospaciari in https://github.com/oven-sh/bun/pull/4503
* fix(node:net): emit close event on connection error by @paperdave in https://github.com/oven-sh/bun/pull/4336
* feat(runtime): Implement `fs.watchFile` by @paperdave in https://github.com/oven-sh/bun/pull/4467
* Align `process.nextTick` execution order with Node by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4409
* fix text decode trim by @dylan-conway in https://github.com/oven-sh/bun/pull/4495
* update root certs by @cirospaciari in https://github.com/oven-sh/bun/pull/4499
* Update development.md by @ThatOneBro in https://github.com/oven-sh/bun/pull/4480
* allocate task for `ThreadSafeFunction`  by @dylan-conway in https://github.com/oven-sh/bun/pull/4513
* fix(runtime): fix `events.once` not using a once listener. by @paperdave in https://github.com/oven-sh/bun/pull/4520
* Update node:fs compat docs cp/cpSync/watchFile/unwatchFile by @birkskyum in https://github.com/oven-sh/bun/pull/4525
* feat(fetch) rejectUnauthorized and checkServerIdentity by @cirospaciari in https://github.com/oven-sh/bun/pull/4514
* Progress for Next.js  by @paperdave in https://github.com/oven-sh/bun/pull/4468
* revert by @paperdave in https://github.com/oven-sh/bun/pull/4539
* fix(fetch) closeAndFail instead of close by @cirospaciari in https://github.com/oven-sh/bun/pull/4537
* fix(cli): final touches for 1.0 by @paperdave in https://github.com/oven-sh/bun/pull/4538
* Doc updates for v1.0 by @colinhacks in https://github.com/oven-sh/bun/pull/4485
* Bump version to Bun 1.0.0 by @paperdave in https://github.com/oven-sh/bun/pull/4543
* Fix typo in sveltekit.md by @hiadamk in https://github.com/oven-sh/bun/pull/4550
* Fix a couple important bugs by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4560

## New Contributors
* @prabhatexit0 made their first contribution in https://github.com/oven-sh/bun/pull/4374
* @JorgeJimenez15 made their first contribution in https://github.com/oven-sh/bun/pull/4445
* @karlbohlmark made their first contribution in https://github.com/oven-sh/bun/pull/4320
* @birkskyum made their first contribution in https://github.com/oven-sh/bun/pull/4525
* @hiadamk made their first contribution in https://github.com/oven-sh/bun/pull/4550

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.8.1...bun-v1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.0'>Bun v1.0</a>.</em>

**Comment by Yakiyo** [comment]
lets gooooo, finally some windows action. incredibly glad to see windows support, even if its experimental.

**Comment by Milo40** [comment]
Yay ! 
It's finally on v1. 🎉 🎉

**Comment by eldoy** [comment]
Congrats! Looks awesome.

**Comment by Jak2k** [comment]
Can I just use bun instead of pnpm+node in my vite-project?

**Reply by birkskyum** [reply]
@Jak2k in many cases yes. Depends what other tools you use in your project, and if you use advanced pnpm features.

**Reply by Jak2k** [reply]
I use the [vitesse-template](https://github.com/antfu/vitesse).

Is it also possible to replace vite with bun while keeping all vite plugins?

**Reply by birkskyum** [reply]
Should be possible yes

**Comment by SwatDoge** [comment]
I was here

**Comment by thenewdesign** [comment]
So stoked for the Windows version of this!!! Congratulations, Jarred

**Comment by SilviaBeatrizRamirez** [comment]
definitivamente  excelente

**Comment by SilviaBeatrizRamirez** [comment]
1. James Madison
1. James Monroe
1. John Quincy Adams
